index.js

```diff
@@ -1,18 +1,25 @@
 const async = require('async')
 const got = require('got')
 
-function heads(urls, callback) {
+function heads(urls, options, callback) {
+  options = options || {};
+
+  if (typeof options === 'function') {
+    callback = options;
+    options = undefined;
+  }
+
   if (typeof urls === 'string') {
     // just one URL
-    get(urls, callback)
+    get(options, urls, callback)
   } else {
     // an array of URLs
-    async.map(urls, get, callback)
+    async.map(urls, get.bind(null, options), callback)
   }
 }
 
-function get(url, callback) {
-  got.head(url)
+function get(options, url, callback) {
```

Comment:
Swapping the arguments was easier and required less changes. Should be fine since this is just an internal function